include/zephyr/posix/signal.h

```diff
@@ -92,12 +102,34 @@ struct sigevent {
 	int sigev_signo;
 };
 
+typedef struct {
+	int si_signo;
+	int si_code;
+	union sigval si_value;
+} siginfo_t;
+
+struct sigaction {
+	void (*sa_handler)(int signno);
+	sigset_t sa_mask;
+	int sa_flags;
+	void (*sa_sigaction)(int signo, siginfo_t *info, void *context);
+};
+
+unsigned int alarm(unsigned int seconds);
+int kill(pid_t pid, int sig);
+int pause(void);
+int raise(int signo);
+int sigaction(int sig, const struct sigaction *ZRESTRICT act, struct sigaction *ZRESTRICT oact);
+int sigpending(sigset_t *set);
+int sigsuspend(const sigset_t *sigmask);
+int sigwait(const sigset_t *ZRESTRICT set, int *ZRESTRICT signo);
 char *strsignal(int signum);
 int sigemptyset(sigset_t *set);
 int sigfillset(sigset_t *set);
 int sigaddset(sigset_t *set, int signo);
 int sigdelset(sigset_t *set, int signo);
 int sigismember(const sigset_t *set, int signo);
+void (*signal(int signo, void (*)(int signo)))(int signo);
```

Comment:
GNU extension is so much clear :)

```
typedef void (*sighandler_t)(int);

sighandler_t signal(int signum, sighandler_t handler);
```

Reply:
It is convenient, but also a gnu-ism.

Reply:
yep :)

Reply:
I haven't seen any conflicts when using the typedef, even when building using `native_sim`, so I'll add it as well.